finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4DispatcherBase.scala

```diff
@@ -105,6 +105,14 @@ trait Netty4DispatcherBase[SendMsg <: Message, RecvMsg <: Message] {
         log.debug(e, "[%s] dispatcher closed", prefix)
         Future.exception(e)
 
+      case Throw(e: ChannelClosedException)
+      // if all streams have already been closed, then this just means that
+      // the client failed to send a GOAWAY frame...
+      if closed.get || streams.keys.asScala.forall { _ <= closedId.get } =>
+        // ...so we don't need to propagate the exception
+        log.warning("[%s] client closed connection without sending GOAWAY frame", prefix)
```

Comment:
(an alternative, and perhaps more principled, solution to this issue: "we submit patches to `grpc-go` and the Python `grpcio` package fixing this behavior, but I think that would take work)